--- glibc-2.4/elf/rtld.c.orig	2008-02-01 16:15:03.997699648 +0300
+++ glibc-2.4/elf/rtld.c	2008-02-01 16:14:57.000000000 +0300
@@ -77,10 +77,12 @@
 char **_dl_argv = NULL;
 /* Nonzero if we were run directly.  */
 unsigned int _dl_skip_args attribute_hidden;
+int _dl_argv0 attribute_hidden = 0;
 #else
 int _dl_argc attribute_relro attribute_hidden;
 char **_dl_argv attribute_relro = NULL;
 unsigned int _dl_skip_args attribute_relro attribute_hidden;
+int _dl_argv0 attribute_relro attribute_hidden = 0;
 #endif
 INTDEF(_dl_argv)
 
@@ -956,6 +958,13 @@
 	    _dl_argc -= 2;
 	    INTUSE(_dl_argv) += 2;
 	  }
+	else if (! strcmp (INTUSE(_dl_argv)[1], "--argv0"))
+	  {
+	    _dl_argv0 = 1;
+	    _dl_skip_args += 1;
+	    _dl_argc -= 1;
+	    INTUSE(_dl_argv) += 1;
+	  }
 	else
 	  break;
 
@@ -981,6 +990,7 @@
                         object we can handle\n\
   --library-path PATH   use given PATH instead of content of the environment\n\
                         variable LD_LIBRARY_PATH\n\
+  --argv0               ARGS-FOR-PROGRAM start from argv[0], not from argv[1]\n\
   --inhibit-rpath LIST  ignore RUNPATH and RPATH information in object names\n\
                         in LIST\n");
 
@@ -1078,6 +1088,12 @@
 	 information for the program.  */
     }
 
+  if (_dl_argv0) {
+	    _dl_skip_args += 1;
+	    _dl_argc -= 1;
+	    INTUSE(_dl_argv) += 1;
+  }
+
   main_map->l_map_end = 0;
   main_map->l_text_end = 0;
   /* Perhaps the executable has no PT_LOAD header entries at all.  */
